integrations/gitlab/gitlab_integration/git_integration.py

```diff
@@ -1,47 +1,125 @@
+import typing
 from functools import lru_cache
 from typing import Dict, Any, Tuple, List
 
-from gitlab_integration.core.entities import FILE_PROPERTY_PREFIX
-from gitlab_integration.gitlab_service import GitlabService
+from gitlab_integration.core.entities import (
+    FILE_PROPERTY_PREFIX,
+    SEARCH_PROPERTY_PREFIX,
+)
+from gitlab_integration.gitlab_service import PROJECTS_CACHE_KEY
 from loguru import logger
 from pydantic import Field
+from gitlab.v4.objects import Project
 
 from port_ocean.context.event import event
 from port_ocean.core.handlers import JQEntityProcessor
 from port_ocean.core.handlers.port_app_config.models import PortAppConfig
 
 
-class GitManipulationHandler(JQEntityProcessor):
+class FileEntityProcessor(JQEntityProcessor):
+    prefix = FILE_PROPERTY_PREFIX
+
+    def _search(self, data: Dict[str, Any], pattern: str) -> Any:
+        project_id, ref = self._validate_project_scope(data)
+        # assuming that the code is being called after event initialization and as part of the GitLab service
+        # initialization
+        project_client: Project | None = event.attributes["PROJECTS_CACHE_KEY"][
+            project_id
+        ]
+        logger.info(f"Searching for file {pattern} in Project {project_id}, ref {ref}")
+
+        file_path = pattern.replace(self.prefix, "")
+        return (
+            self._get_file_content(project_client, file_path, ref)
+            if project_client
+            else None
+        )
+
     @lru_cache()
     def _get_file_content(
```

Comment:
Im not sure the lru is working anymore because ur recreating the class each time